client/state/login/reducer.js

```diff
@@ -56,6 +59,9 @@ export const redirectTo = createReducer( null, {
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST ]: () => null,
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST_FAILURE ]: ( state, action ) => get( action, 'redirectTo', null ),
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST_SUCCESS ]: () => null,
+	[ LOGOUT_REQUEST ]: () => null,
+	[ LOGOUT_REQUEST_SUCCESS ]: () => ( state, { data } ) => get( data, 'redirect_to', null ),
+	[ LOGOUT_REQUEST_FAILURE ]: () => null,
 } );
```

Comment:
We kind of treat changing the current user state - either logging in or out - as something that requires reloading the page to ensure the correct data is in the state tree. Resetting the `login` subtree is probably overkill when it's going to be reset when the page reloads.